File: models/Doctor.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const doctorSchema = new Schema(
  {
    firstName: {
      type: String,
      required: [true, "Please enter a first name"],
    },
    lastName: {
      type: String,
      required: [true, "Please enter a last name"],
    },
    username: {
      type: String,
      required: [true, "Please enter a username"],
    },
    password: {
      type: String,
      required: [true, "Please enter a password"],
    },
    email: {
      type: String,
      required: [true, "Please enter an email"],
    },
    dateOfBirth: {
      type: String,
      required: [true, "Please enter a Date of Birth"],
    },
    hourlyRate: {
      type: Number,
      required: [true, "Please enter an hourly rate"],
    },
    affiliation: {
      type: String,
      required: [true, "Please enter an affiliation"],
    },
    degree: {
      type: String,
      required: [true, "Please enter a degree"],
    },
    status: {
      type: String,
      enum: ["approved", "pending", "contract", "rejected"],
      default: "Pending",
      required: false,
    },
    speciality: {
      type: String,
      required: [true, "Please enter a specialty"],
    },
    patients: {
      type: [String],
      required: false,
    },
    wallet: {
      type: Number,
      default: 0,
      required: false,
    },
    socketId: { type: String, required: false },
    idDocument: { type: String },
    medicalDegree: { type: String },
    medicalLicense: { type: String },
    messages: {
      type: [
        {
          sender: String,
          content: String,
          timestamp: { type: Date, default: Date.now },
        },
      ],
      default: [],
      required: false,
    },
  },
  { timestamps: true }
);

const Doctor = mongoose.model("Doctor", doctorSchema, "Doctors");
module.exports = Doctor;

File: clinic_front_end/lib/Models/Doctor.dart
class Doctor {
  String id = "";
  String firstName = "";
  String lastName = "";
  String username = "";
  String password = "";
  String email = "";
  String dateOfBirth = "";
  int hourlyRate = 0;
  String affiliation = "";
  String speciality = "";
  String degree = "";
  String status = "";
  List<String> patients = [];
  List<String> appointments = [];
  double price = 0;

  Doctor.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    firstName = json['FirstName'];
    lastName = json['LastName'];
    username = json['Username'];
    password = json['Password'];
    email = json['Email'];
    dateOfBirth = json['DateOfBirth'];
    hourlyRate = json['HourlyRate'];
    affiliation = json['affiliation'];
    speciality = json['Specialty'] ?? "";
    degree = json['Degree'];
    status = json['Status'];
    for (var item in json['Patients']) {
      patients.add(item.toString());
    }
    for (var item in json['Appointments']) {
      appointments.add(item.toString());
    }
  }
}
